mailbox/backup/src/test/java/org/apache/james/mailbox/backup/DefaultMailboxBackupTest.java

```diff
@@ -0,0 +1,225 @@
+            EntryChecks.hasName(MESSAGE_ID_1.serialize()).hasStringContent(MESSAGE_CONTENT_1)
+        );
+    }
+
+    @Test
+    void doBackupShouldOnlyArchiveTheMailboxOfTheUser() throws Exception {
+        MailboxSession session = mailboxManager.createSystemSession(USER);
+        MailboxSession otherSession = mailboxManager.createSystemSession(OTHER_USER);
+
+        createMailBoxWithMessage(session, MAILBOX_PATH_USER1_MAILBOX1, MESSAGE_1);
+        createMailBoxWithMessage(otherSession, MAILBOX_PATH_OTHER_USER_MAILBOX1, MESSAGE_1_OTHER_USER);
+
+        Publisher<BlobId> res = backup.backupAccount(USER1);
+
+        assertThatZipFromIdContainsOnly(res,
+            EntryChecks.hasName(MAILBOX_1_NAME + "/").isDirectory(),
+            EntryChecks.hasName(MESSAGE_ID_1.serialize()).hasStringContent(MESSAGE_CONTENT_1)
+        );
+    }
+
+    @Test
+    void backupEmptAccountThenRestoringItInUser2AccountShouldCreateNoElements() throws Exception {
+        Publisher<BlobId> backupRes = backup.backupAccount(USER1);
+        Mono.from(backupRes).flatMap(Throwing.function(blobId -> Mono.from(backup.restore(USER2, blobId)))).block();
+
+        MailboxSession sessionUser2 = mailboxManager.createSystemSession(OTHER_USER);
+        List<DefaultMailboxBackup.MailAccountContent> content = backup.getAccountContentForUser(sessionUser2);
+
+
+        assertThat(content).isEmpty();
+    }
+
+    @Test
+    void backupAccountWithOneMailboxThenRestoringItInUser2AccountShouldCreateOneMailbox() throws Exception {
+        MailboxSession sessionUser1 = mailboxManager.createSystemSession(USER);
+        createMailBoxWithMessage(sessionUser1, MAILBOX_PATH_USER1_MAILBOX1);
+
+        Publisher<BlobId> backupRes = backup.backupAccount(USER1);
+        Mono.from(backupRes).flatMap(Throwing.function(blobId -> Mono.from(backup.restore(USER2, blobId)))).block();
+
+        MailboxSession sessionUser2 = mailboxManager.createSystemSession(OTHER_USER);
+        List<DefaultMailboxBackup.MailAccountContent> content = backup.getAccountContentForUser(sessionUser2);
+
+        assertThat(content).hasSize(1);
+        DefaultMailboxBackup.MailAccountContent mailAccountContent = content.get(0);
+        Mailbox mailbox = mailAccountContent.getMailboxWithAnnotations().mailbox;
+        assertThat(mailbox.getName()).isEqualTo(MAILBOX_1_NAME);
+        assertThat(mailAccountContent.getMessages().count()).isEqualTo(0);
+    }
+
+    @Test
+    void backupAccountWithOneMailboxAndTwoMessageThenRestoringItInUser2AccountShouldCreateOneMailboxWithTwoMessage() throws Exception {
+        MailboxSession sessionUser1 = mailboxManager.createSystemSession(USER);
+        createMailBoxWithMessage(sessionUser1, MAILBOX_PATH_USER1_MAILBOX1, MESSAGE_1, MESSAGE_2);
+
+        Publisher<BlobId> backupRes = backup.backupAccount(USER1);
+        Mono.from(backupRes).flatMap(Throwing.function(blobId -> Mono.from(backup.restore(USER2, blobId)))).block();
```

Comment:
Hard to understand...

Please again use blobId directly. Also use method extraction to clearly expose the action performed:

```
BlobId backUp(User user)

+ 

void restore(User user, BlobId blobId)
```

Will help

also organise the complicated tests into Given/When/Then blocks separated by blank lines is generaly a huge gain